### DISCUSSION TITLE
How to ignore rules in shareable config if plugin is not installed?

### DISCUSSION THREAD

**Discussion by aivchen** [eslint/eslint#17586] [Q&A] [answered]
Hi!
I'm working on a shareable config which will be used in all projects in our organization.
Projects are very different. For example, in one of them we use Vue. But in another one we use vanilla JS.

The idea is to create single config with all rules. For example:
```js
module.exports = {
  extends: 'eslint:recommended',
  rules: {
    'semi': ['error', 'never'],
    'vue/order-in-components': 'error',
  }
}
```

And package.json for this config:
```json
{
  "peerDependencies": {
    "eslint": "^7.29.0 || ^8.26.0",
    "eslint-plugin-vue": "^9.17.0"
  }
}
```

But now in the target project where we don't use Vue I get errors like this:
```
Definition for rule 'vue/order-in-components' was not found
```

Obviously, it's due to missed `eslint-plugin-vue`. But we don't need vue checks in this project at all.

So the question is there any way to specify in the shareable config to not check for all vue/* rules if `eslint-plugin-vue` is not installed in the target project? 
Thank you!

**Comment by nzakas** [comment, marked as answer]
> So the question is there any way to specify in the shareable config to not check for all vue/* rules if eslint-plugin-vue is not installed in the target project?

No. We always perform this validation to prevent people from configuring rules that aren't loaded. The only time we don't check if a rule definition exists is if it's set to "off", so `vue/order-in-components: "off"` would not trigger the check.

**Reply by aivchen** [reply]
Thank you so much!
